### ISSUE TITLE
First-class external-validation node (documented predict-only kit, metric-only reporting)

### ISSUE THREAD

**Issue by Ultimate-Storm** [KatherLab/MediSwarm#412]
## Problem
There's no first-class way for a center to join **purely for external validation** (predict-only, no training). The capability exists but is ad-hoc: `scripts/evaluation/predict.py --checkpoint FL_global_model.pt` run via the internal smoke/deploy script ([run_startupkit_smoke_eval_dl0.sh:177](scripts/deploy/run_startupkit_smoke_eval_dl0.sh#L177)), with `prediction_flag='ext'` ([env_config.py:37](application/jobs/_shared/custom/env_config.py#L37)). A new site currently has no clean, documented workflow to validate a released model on their own data without joining the swarm.

## Fix — a documented "external-validation kit" (predict-only node)
- [ ] Provide a **predict-only startup kit / provisioning profile**: no swarm membership, no VPN/aggregation path — just the odelia image + the released global model + a wrapper around `predict.py`.
- [ ] Workflow: site prepares data with a **`test`/`ext` split**, passes the **preflight** (the same data-integrity/GPU guard that caught CAM's 11 corrupt inputs), runs `predict.py --checkpoint best_FL_global_model.pt`, and gets metrics + per-sample predictions **locally**.
- [ ] **Metric-only reporting**: only aggregate metrics leave the site (predictions stay local) — privacy-preserving external validation.
- [ ] Add operator docs (a section in `assets/readme/` / `docs/`) + a one-command wrapper (`scripts/deploy/run_external_validation.sh`).
- [ ] Reuse the extended metric set from the evaluation-metrics issue so external validation is directly comparable to the training-site numbers.

## Verification
- [ ] A node with only {data + released global model + the predict-only kit} runs external validation end-to-end **without** joining the swarm.
- [ ] Outputs `prediction_results.json` + `predictions_*.csv`; only metrics need to be shared back.
- [ ] Preflight rejects a corrupt/degenerate dataset before predicting.

**Comment by Ultimate-Storm** [comment]
## Implementation plan (requires kit regeneration to verify)

### Most of this already exists
`scripts/evaluation/predict.py` is a **complete standalone predict/eval tool**:
- mutually-exclusive `--checkpoint <path...>` or `--workspace <dir>`; plus `--model-name`, `--checkpoint-type {state_dict,lightning}`, `--ensemble`, `--best-only`, `--output-dir`, `--split {train,val,test}`, `--batch-size`, `--device`, `--num-classes`
- loads NVFlare `FL_global_model.pt` / `best_FL_global_model.pt` state-dicts
- writes per-sample CSV + `prediction_results.json` + console ACC / AUROC / F1 / confusion matrix
- reads `DATA_DIR` / `SITE_NAME` / `SCRATCH_DIR` / `MODEL_NAME`; builds data via `ODELIA_Dataset3D`

And `scripts/deploy/run_startupkit_smoke_eval_dl0.sh:160-186` **already runs it predict-only** against a global model via a hand-written `docker run` (data mounted `:ro`, checkpoint dir `:ro`, output rw).

### The actual gap
The **generated startup kit's `docker.sh`** (from `docker_config/master_template.yml`) has **no predict/eval mode**. Its arg parser (`:719-736`) and usage block (`:1032-1058`) offer only:
`--data_dir --scratch_dir --no_pull --dummy_training --preflight_check --local_training --start_client --list_licenses --interactive --run_script --job --log_dataset_details --model_name --num_epochs --config --container_name`

So today an external centre can only evaluate via `--run_script` / `--interactive` or a bespoke `docker run`. There is no supported, documented path.

### Plan
- [ ] Add an `--external_validation` (alias `--predict`) mode to `master_template.yml`'s `docker_cln_sh` template:
  - new flags `--checkpoint <path>` (required) and optional `--split`, `--output-dir`, `--ensemble`, `--best-only`
  - mounts the supplied global model **read-only**, data **read-only**, output rw
  - invokes `python3 /MediSwarm/scripts/evaluation/predict.py …` (reuse it verbatim — do not reimplement)
  - add the mode to the usage block
- [ ] Reuse the exact `docker run` invocation already proven in `run_startupkit_smoke_eval_dl0.sh:160-186`.
- [ ] Ship a **predict-only kit** variant: no client cert / no swarm join, no VPN requirement — the node never participates in training.
- [ ] **Metric-only reporting**: the external node returns `prediction_results.json` (metrics) only; per-sample predictions and weights stay on site. Document this explicitly — it is the governance argument for letting a non-consortium centre validate.
- [ ] Document in `docs/` : how a centre obtains the global model, runs one command, and reports metrics back.

### Verification (needs kit regeneration)
- [ ] `./docker.sh --external_validation --checkpoint FL_global_model.pt --data_dir … --scratch_dir …` produces `prediction_results.json` + metrics, writes nothing outside `--output-dir`.
- [ ] Fails clearly when `--checkpoint` is missing or unreadable.
- [ ] The predict-only kit contains no client credentials and never contacts the FL server.
- [ ] Numbers match a `--run_script predict.py` run on the same checkpoint/data (no behavior drift).

### Related
Pairs naturally with #410 (broadened metrics) — an external validator reporting sensitivity/specificity/PR-AUC/CIs rather than just ACC/AUROC is far more useful. `predict.py` currently computes its own metrics via sklearn, so it should be updated to report the same expanded set.
